src/utils.js

```diff
@@ -34,6 +35,33 @@ module.exports.mkDirPromise = (dirPath) => (
   })
 );
 
+module.exports.readDirPromise = dirPath => (
+  new Promise((resolve, reject) => {
+    fs.readdir(dirPath, 'utf-8', (err, files) => {
+      err ? reject(err) : resolve(files);
+    });
+  })
+);
+
+// Simple promise wrapper for rimraf module
+module.exports.rimrafPromise = dirPath => (
+  new Promise((resolve, reject) => {
+    rimraf(dirPath, (err) => {
+      err ? reject(err) : resolve();
+    });
+  })
+);
+
+
+module.exports.statPromise = dirPath => (
+  new Promise((resolve, reject) => {
+    fs.stat(dirPath, (err, file) => {
+      err ? reject(err) : resolve(file);
+    });
+  })
+);
```

Comment:
Thanks for these helpers!!

If you wanted to be really fancy, you could make a "promise wrapper" helper. Something like:

```js
const wrapInPromise = fn => (...args) => (
  new Promise((resolve, reject) => {
    fn(...args, (err, ...data) => {
      err ? reject(err) : resolve(...data);
    });
  })
);

module.exports.readDirPromise = wrapInPromise(fs.readdir);
module.exports.rimrafPromise = wrapInPromise(rimraf);
module.exports.statPromise = wrapInPromise(fs.stat);
```

You don't have to do this if you don't want though :) I'm not sure it's actually helpful, may be too clever for its own good. Plus then you'd have to specify the character encoding as UTF-8 when you use `readDirPromise`, since there's no way to automatically provide it.

(Also I'm not sure if Node natively supports rest/spread, so it may be impossible anyway?)